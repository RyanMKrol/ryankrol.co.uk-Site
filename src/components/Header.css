.Header {
  height: 30%;
  width: 100%;
  background-color: #FDD023;
  position: relative;
}

.links {
  width: 90%;
  height: 100%;
  bottom: 0;
  margin: 0 auto;
  background-color: black;
  color: white;
}

.links-container {
  width: 100%;
  height: 30%;
  position: absolute;
  bottom: 0;
}
