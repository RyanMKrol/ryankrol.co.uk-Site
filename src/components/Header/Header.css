.Header {
  height: 240px;
  width: 100%;
  background-color: #FDD023;
  display: flex;
  border-bottom-width: 3px;
  border-bottom-style: solid;
  border-bottom-color: #FDD023;
}

.header-name {
  font-family: 'Bungee Shade', cursive;
  display: flex;
  align-items: center;
  font-size: 76px;
  justify-content: center;
  flex-grow: 1;
}

@media screen and (max-width: 600px) {
  .header-name {
    font-size: 46px;
  }
}

.header-content {
  width: 90%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
}

.social-container {
  display: flex;
}

.links {
  height: 100%;
  width:100%;
  bottom: 0;
  background-color: #555;
  overflow-x: scroll;
}

.links-container {
  width: 100%;
  height: 30%;
  align-self: flex-end;
}
.links-inner-container {
  width: 100%;
  height: 100%;
  vertical-align: middle;
  display: flex;
}

.header-link {
  margin: auto 0;
  padding: 0 20px;

  position: relative;

  font-family: 'Bebas Neue', cursive;
  font-size: 22px !important;

  color: #FDD023;
  outline: none;
  text-decoration: none;
}

/* Styles around the underlining present on links in the header */
.header-link:before, .currentPage:before {
  content: '';
  position: absolute;

  border-bottom: 1.5px solid #FDD023;

  padding: 0 3px;
  margin: -3px auto;

  width: 0;
  bottom: 0;
  left: 0;
  right: 0;

  transition: 0.4s;
}

/* Transition to a wide border */
.header-link:hover:before, .currentPage:before {
  width: 80%;
}
