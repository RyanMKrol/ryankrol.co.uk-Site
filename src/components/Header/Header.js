import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';

import styles from './Header.module.css';

const HEADER_ITEMS = [
  {
    destination: '/ReadingList',
    title: 'reading list',
  },
  {
    destination: '/MoviesList',
    title: 'movies',
  },
  {
    destination: '/AlbumRatings',
    title: 'album ratings',
  },
  {
    destination: '/MovieRatings',
    title: 'movie ratings',
  },
];

/**
 * Internal class for rendering a header item
 */
class HeaderItem extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.state = {
      isHovered: false,
    };

    this.handleHover = this.handleHover.bind(this);
  }

  /**
   * Handles hover events
   */
  handleHover() {
    this.setState((prevState) => ({
      isHovered: !prevState.isHovered,
    }));
  }

  /**
   * Renders the component
   *
   * @returns {React.Component} JSX for individual header link
   */
  render() {
    const btnClass = this.state.isHovered ? 'highlighted' : '';
    return (
      <Link
        className={`${btnClass} light ${styles.link}`}
        onMouseEnter={this.handleHover}
        onMouseLeave={this.handleHover}
        to={this.props.destination}
      >
        <span className="highlighted">{this.props.index}. </span>
        {this.props.content}
      </Link>
    );
  }
}

/**
 * Generates the header links
 *
 * @returns {React.Component} JSX for link items
 */
const generateHeaderLinks = () => HEADER_ITEMS.map((linkItem, index) => (
    <HeaderItem
      key={linkItem.destination}
      content={linkItem.title}
      destination={linkItem.destination}
      index={index + 1}
    />
));

/**
 * Component for rendering common page heading
 *
 * @returns {React.Component} JSX Header component
 */
function Header() {
  return (
    <div className={styles.outer}>
      <div className={styles.container}>
        <Link className={`${styles.logo} highlighted`} to={'/'}>
          RK
        </Link>
        <div className={styles.links}>{generateHeaderLinks()}</div>
      </div>
    </div>
  );
}

export default withRouter(Header);
