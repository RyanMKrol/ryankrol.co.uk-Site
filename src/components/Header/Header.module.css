.container {
  display: flex;
  width: 100%;
}

.logo {
  color: red;
}

.links {
  display: flex;
  margin-left: auto;
}
