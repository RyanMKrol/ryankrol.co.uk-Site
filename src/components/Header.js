import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import SocialList from './SocialList';
import { HEADER_ITEMS } from '../data/HeaderData';

import './Header.css';

/**
 * Object representing the location of the page
 *
 * @typedef {object} LocationObject
 * @property {string} pathname What URL the page is currently on
 */

/**
 * Object representing the props to the Header
 *
 * @typedef {object} HeaderProps
 * @property {LocationObject} location Object representing the current URL
 */

/**
 * Class representing a UI component for an album
 */
class Header extends Component {
  /**
   * Generates the links used in the header
   *
   * @returns {string} HTML markup for the header link
   */
  generateHeaderLinks() {
    return HEADER_ITEMS.map((linkItem) => {
      const activeClass = linkItem.destinationPage === this.props.location.pathname ? 'currentPage' : '';

      return (
        <Link
          key={linkItem.destinationPage}
          className={`${activeClass} header-link hover`}
          to={linkItem.destinationPage}
        >
          {linkItem.content}
        </Link>
      );
    });
  }

  /**
   * Render method
   *
   * @returns {React.Component} The JSX representing the component
   */
  render() {
    return (
      <div className="Header">
        <div className="header-content">
          <div className="social-container">
            <SocialList />
          </div>
          <div className="header-name">Ryan Krol</div>
          <div className="links-container">
            <div className="links">
              <div className="links-inner-container">{this.generateHeaderLinks()}</div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(Header);
