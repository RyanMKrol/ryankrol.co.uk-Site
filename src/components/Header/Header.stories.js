import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from './Header';

export default {
  title: 'Components/Header',
};

export const Default = () => (
  <Router>
    <Header />
  </Router>
);
