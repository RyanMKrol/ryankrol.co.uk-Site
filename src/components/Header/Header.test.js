import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

it('renders header correctly', () => {
  const header = shallow(<Header.WrappedComponent location={{}} />);
  expect(toJson(header)).toMatchSnapshot();
});

it('renders header without throwing', () => {
  expect(() => mount(
      <Router>
        <Header />
      </Router>,
  )).not.toThrow();
});
