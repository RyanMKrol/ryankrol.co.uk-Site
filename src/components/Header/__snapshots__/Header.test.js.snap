// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
<div
  className="outer"
>
  <div
    className="container"
  >
    <Link
      className="logo highlighted"
      to="/"
    >
      RK
    </Link>
    <div
      className="links"
    >
      <HeaderItem
        content="Movie Ratings"
        destination="/ratings/movie"
        index={1}
        key="/ratings/movie"
      />
      <HeaderItem
        content="Television Ratings"
        destination="/ratings/tv"
        index={2}
        key="/ratings/tv"
      />
      <HeaderItem
        content="Album Ratings"
        destination="/ratings/album"
        index={3}
        key="/ratings/album"
      />
      <HeaderItem
        content="Books"
        destination="/reading"
        index={4}
        key="/reading"
      />
      <HeaderItem
        content="Movies"
        destination="/movies"
        index={5}
        key="/movies"
      />
    </div>
  </div>
</div>
`;
